src/PIL/IcoImagePlugin.py

```diff
@@ -142,11 +142,8 @@ def __init__(self, buf):
             msg = "No images were found"
             raise TypeError(msg)
 
-        self.entry = sorted(self.entry, key=lambda x: x["color_depth"])
-        # ICO images are usually squares
-        # self.entry = sorted(self.entry, key=lambda x: x['width'])
-        self.entry = sorted(self.entry, key=lambda x: x["square"])
-        self.entry.reverse()
```

Comment:
I presume you dropped the first sort because you think it is necessary with the second sort? To me, it is still necessary in the scenario where two entries have the same "square" value.

For a simple example,
```python
x = [9,8,7,6,5,4,3,2,1]
x.sort(key=lambda x: x // 2)
print(x)
```
does not produce the same output as
```python
x = [9,8,7,6,5,4,3,2,1]
x.sort(key=lambda x: x % 2)
x.sort(key=lambda x: x // 2)
print(x)
```

Reply:
Ah, yeah, you're right (and this ties to the fact that `.sort()` is guaranteed stable).

The intent of the original code has been to sort so the most color-rich, largest image is preferred, but that's not clear from the comments (and commented-out old code). I'll add `color_depth` back to the sort key.